work_queue/src/work_queue.c

```diff
@@ -7413,4 +7412,295 @@ int work_queue_specify_min_taskid(struct work_queue *q, int minid) {
 	return q->next_taskid;
 }
 
+//the functions below are used by qsort in order to sort the work_queue_wsummary data
+int comparecount(const void *a, const void *b)
+{
+	return ( ((struct work_queue_bucket *)b)->count - ((struct work_queue_bucket *)a)->count );
+}
+int comparecores(const void *a, const void *b)
+{
+	return ( ((((struct work_queue_bucket *)b)->cores - ((struct work_queue_bucket *)a)->cores) == 0) ? 
```

Comment:
Stylistic simplification:
Let `struct work_queue_bucket *x = a, *y=b` then you can use `x` and `y` instead of all the casts below.